core/src/main/scala/org/apache/spark/deploy/rest/mesos/MesosRestServer.scala

```diff
@@ -99,7 +99,7 @@ private[mesos] class MesosSubmitRequestServlet(
     // cause spark-submit script to look for files in SPARK_HOME instead.
     // We only need the ability to specify where to find spark-submit script
     // which user can user spark.executor.home or spark.home configurations.
-    val environmentVariables = request.environmentVariables.filter(!_.equals("SPARK_HOME"))
+    val environmentVariables = request.environmentVariables.filterKeys(!_.equals("SPARK_HOME"))
```

Comment:
Yeah, that'd be ok too. I chose to leave the code as it was and do the minimal change (given that we are in the RC cycle)